src/Silverback.Integration/Diagnostics/ActivityExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) 2019 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Diagnostics
+{
+    internal static class ActivityExtensions
+    {
+        public static void AddBaggageRange(this Activity activity, IEnumerable<KeyValuePair<string, string>> baggageItems)
+        {
+            foreach (var baggageItem in baggageItems)
+            {
+                activity.AddBaggage(baggageItem.Key, baggageItem.Value);
+            }
+        }
+
+        public static void FillActivity(this Activity activity, IBrokerMessage brokerMessage)
+        {
+            brokerMessage.Headers.Add(new MessageHeader(DiagnosticsConstants.TraceIdHeaderKey, activity.Id));

```

Comment:
Check activity.Id for null and throw an InvalidOperationException if its null. The activityId is null, if there is no started activity. As it makes no sense to call "FillActivity" in this case, this is a behavior which should never happen. Thus, the exception notifies the dev that he uses the code the wrong way.